pkg/cvo/availableupdates.go

```diff
@@ -432,36 +429,69 @@ func unknownExposureMessage(risk configv1.ConditionalUpdateRisk, err error) stri
 	return fmt.Sprintf(template, risk.Name, err, risk.Name, risk.Message, risk.Name, risk.URL)
 }
 
-func evaluateConditionalUpdate(ctx context.Context, conditionalUpdate *configv1.ConditionalUpdate, conditionRegistry clusterconditions.ConditionRegistry) *metav1.Condition {
-	recommended := &metav1.Condition{
-		Type: ConditionalUpdateConditionTypeRecommended,
+func newRecommendedStatus(now, want metav1.ConditionStatus) metav1.ConditionStatus {
+	switch {
+	case now == metav1.ConditionFalse || want == metav1.ConditionFalse:
+		return metav1.ConditionFalse
+	case now == metav1.ConditionUnknown || want == metav1.ConditionUnknown:
+		return metav1.ConditionUnknown
+	default:
+		return want
 	}
-	messages := []string{}
-	for _, risk := range conditionalUpdate.Risks {
+}
+
+const (
+	recommendedReasonAsExpected       = "AsExpected"
+	recommendedReasonEvaluationFailed = "EvaluationFailed"
+	recommendedReasonMultiple         = "MultipleReasons"
+
+	// recommendedReasonRiskApplies is used instead of the original name if it
+	// does not match the pattern for a valid k8s condition reason.
+	recommendedReasonRiskApplies = "RiskAppliesToCluster"
+)
+
+var reasonPattern = regexp.MustCompile(`^[A-Za-z]([A-Za-z0-9_,:]*[A-Za-z0-9_])?$`)
```

Comment:
This uses https://github.com/openshift/api/blob/59fa376de7cb668ddb95a7ee4e9879d7f6ca2767/vendor/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L1535 , to ensure we'll fit into `conditionalUpdates[].conditions`.  Possibly worth calling out that sourcing in a Go comment or commit message?

Reply:
sounds good, will amend

/hold

Reply:
Added in an amend